### PR TITLE
Redmine #7961: Remove auto-loading of "inputs" mentioned in def.json.

### PR DESCRIPTION
**Pull request by kacf**
This was done in order to solve the problem that def.json is applied
to all policy files, including failsafe.cf. If there is a syntax error
in any of those files, all policy files will stop working and the host
is unable to recover.

Now one has to use the "inputs" section by referring to it as the
$(def.augments_inputs) variable, which it will populate.

Changelog: Fix bug which could render host unable to recover from a
syntax error, even if failsafe.cf was utilized. This could happen if
the file containing the syntax error was specified in the def.json
special file.

Changelog: Change: Policy files specified in the "inputs" section of
def.json will no longer be auto-loaded. One has to refer to the
$(def.augments_inputs) variable in the policy (the standard
masterfiles policies include this by default). This only affects
installations which are not based on the standard masterfiles, and
which are using the "inputs" field inside def.json.

### CODE REVIEW

**Comment by kacf** [comment]
Do not merge yet! This is missing tests and documentation.

**Comment by kacf** [comment]
@tzz: Do you mind taking a look at this? I'm still missing tests and changing the documentation, but this is a beginning.

File: libpromises/rlist.c [hunk 1]
```diff
@@ -1438,6 +1437,71 @@ JsonElement *RvalToJson(Rval rval)
...
+
+    Rval rval;
+
+    switch (JsonGetElementType(json))
+    {
+    case JSON_ELEMENT_TYPE_CONTAINER:
+        switch (JsonGetContainerType(json))
+        {
+        case JSON_CONTAINER_TYPE_OBJECT:
+            rval.item = json;
```
Comment (kacf) [inline libpromises/rlist.c]: Actually, I probably have to `JsonCopy` this.

File: libpromises/rlist.c [hunk 2]
```diff
@@ -1438,6 +1437,71 @@ JsonElement *RvalToJson(Rval rval)
...
+        }
+
+        default:
+            ProgrammingError("JsonElement container with unknown type");
+            break;
+        }
+        break;
+
+    case JSON_ELEMENT_TYPE_PRIMITIVE:
+        rval.item = JsonPrimitiveGetAsString(json);
```
Comment (kacf) [inline libpromises/rlist.c]: And `xstrdup` this.

**Comment by tzz** [comment]
Looks nice.

**Comment by kacf** [comment]
Test verifying that it works with masterfiles: https://github.com/cfengine/masterfiles/pull/668

**Comment by kacf** [comment]
Alright, updated. I also found `ContainerToRlist` which means I could remove the whole `JsonToRval` function I made. No need for duplication.

**Comment by kacf** [comment]
trigger build

**Comment by kacf** [comment]
Cherry-picked to 3.8.x, 3.7.x is pending.

**Pull request merged**